packages/core/react/src/useAnchorWallet.ts

```diff
@@ -13,7 +13,13 @@ export function useAnchorWallet(): AnchorWallet | undefined {
     return useMemo(
         () =>
             publicKey && signTransaction && signAllTransactions
-                ? { publicKey, signTransaction, signAllTransactions }
+                ? {
+                      publicKey,
+                      signTransaction: signTransaction as (transaction: Transaction) => Promise<Transaction>,
+                      signAllTransactions: signAllTransactions as (
+                          transactions: Transaction[]
+                      ) => Promise<Transaction[]>,
```

Comment:
This is a bit of a hack but should be fine -- we know Anchor's interface doesn't handle `VersionedTransaction` yet, so even if the adapter does, we are fine only allowing `Transaction`.

cc @armaniferrante